sys/targets/targets.go

```diff
@@ -775,6 +778,60 @@ func initTarget(target *Target, OS, arch string) {
 		target.ExecutorUsesForkServer = false
 		target.HostFuzzer = true
 	}
+	target.initAddr2Line()
+}
+
+func (target *Target) initAddr2Line() {
+	// Initialize addr2line lazily since lots of tests don't need it,
+	// but we invoke a number of external binaries during addr2line detection.
+	var (
+		init sync.Once
+		bin  string
+		err  error
+	)
+	target.Addr2Line = func() (string, error) {
+		init.Do(func() { bin, err = target.findAddr2Line() })
+		return bin, err
+	}
+}
+
+func (target *Target) findAddr2Line() (string, error) {
+	// First try to find llvm-addr2line as it's significantly faster on large binaries.
+	if path, err := exec.LookPath("llvm-addr2line"); err == nil {
+		return path, nil
+	}
+	// It may also be installed only as a versioned binary,
+	// the current syzbot container has only llvm-addr2line-15.
+	for ver := 100; ver > 10; ver-- {
+		if path, err := exec.LookPath(fmt.Sprintf("llvm-addr2line-%v", ver)); err == nil {
+			return path, nil
+		}
+	}
+	addr2line := "addr2line"
+	if target.Triple != "" {
+		addr2line = target.Triple + "-" + addr2line
+	}
+	path, err := exec.LookPath(addr2line)
+	if err != nil {
+		return "", errors.New("addr2line is not installed")
+	}
+	if target.OS != Darwin || target.Arch != AMD64 {
```

Comment:
This may change the behavior for `OS == Darwin` -- we now skip all the checks below if `llvm-addr2line` is present. Earlier we always ran them for Darwin. Was this change indended?

Reply:
Restored.